lib/internal/fs/glob.js

```diff
@@ -54,6 +70,28 @@ class Cache {
   addToStatCache(path, val) {
     this.#statsCache.set(path, val);
   }
+  async* opendir(path) {
+    const cached = this.#opendirCache.get(path);
+    if (cached) {
+      yield* await cached;
+      return;
+    }
+    const { promise, resolve } = createDeferredPromise();
+    this.#opendirCache.set(path, promise);
+    let val;
+    try {
+      val = await opendir(path);
+    } catch {
+      resolve([]);
+      return;
+    }
+    const cache = [];
+    for await (const item of val) {
+      yield item;
+      ArrayPrototypePush(cache, item);
+    }
+    resolve(cache);
```

Comment:
A “safe” version would look like:
```suggestion
    resolve({ __proto__: null, [SymbolAsyncIterator]: () => new SafeArrayIterator(cache) });
```
However, it’s not impossible that it would make the code slower than if there wasn’t a cache at all, I haven’t measured.